src/assets/RestApi/rest-api.md

```diff
@@ -0,0 +1,400 @@
+- To retrieve all tasks:
+```python
+@app.route("/tasks", methods=["GET"])
+def get_tasks():
+    return jsonify({"tasks": tasks}), 200
+```
+- The `@app.route` decorator provided by Flask binds a URL pattern to the function that follows it. In this case, the URL pattern in `/tasks`.
+- `methods=["GET]` specifies that this route should respond to the HTTP GET requests. This means that when a client sends a GET request to the `/tasks` URL, Flask will call the associated function.
+- `get_tasks()` function will return a JSON response containing all tasks.
+- `jsonify({"tasks": tasks})` is a Flask function that converts the Python dictionary `{"tasks": tasks}` into a JSON response.
+
+- To retrieve a specific task:
+```python
+@app.route("/tasks/<int:task_id>", methods=["GET"])
+def get_task(task_id):
+    task = next((item for item in tasks if item["id"] == task_id), None)
+    if task is None:
+        return jsonify({"error": "Task not found"}), 404
+    return jsonify({"task": task}), 200
+```
+- `@app.route` binds a URL pattern `/tasks/<int:task_id>` to the function that follows it.
+- `<int:task_id>` is a variable part of the URL that will be captured and passed to the function as an argument.
+- `methods=["GET"]` specifies that this route should respond to the HTTP GET requests. This means that when a client sends a GET request to the `tasks/<int:task_id>` URL, Flask will call the associated function.
+- `get_task()` function will take `task_id` as argument that is extracted from `/tasks/<int:task_id>` URL. The function will then search for a task with that ID in the `tasks` list. If a task is found, it will return the task in JSON format. If no task is found, it will return a `404` status code with an error message.
+- `return jsonify({"task": task})` returns a JSON response containing the task. 
+
+### POST Request
+A POST request is used to send data to a server to create a new resource. To create a new task, implement a route that handles POST requests:
+```python
+@app.route("/tasks", methods=["POST"])
+def create_task():
+    new_task = request.get_json()
+    tasks.append(new_task)
+    return jsonify({"task": new_task}), 201
+```
+- `@app.route` binds a URL pattern `/tasks` to the function that follows it.
+- `methods=["POST"]` specifies that this route should respond to the HTTP POST requests. This means when a client sends a POST request to the `/tasks` URL, Flask will call the associated function.
+- `request.get_json()` is a Flask function that parses the incoming request data as JSON. This function is used to extract the JSON payload sent by the client in the POST request.
+- `create_task()` function will parse the JSON payload, add the new task to the `tasks` list, and return a JSON response containing the newly created task. The status code `201` indicates that the task was successfully created.
+- `tasks.append(new_task)` appends the new_task dictionary to the tasks list.
```

Comment:
```suggestion
- `tasks.append(new_task)` appends the `new_task` dictionary to the tasks list.
```
I think it's a bit clearer if you put the name of the dictionary into quotation marks as you've done above for the list